method.rb

```diff
@@ -0,0 +1,190 @@
+module Enumerable
+  def my_each
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return 'no block given' unless block_given?
+
+    output_array = []
+    i = 0
+    while i < size
+      output_array << self[i] if (yield i) == true
+      i += 1
+    end
+    output_array
+  end
+
+  def my_all?
+    return 'no block given' unless block_given?
```

Comment:
* [ ] There should be a working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
 #### compare the following examples by replacing the custom method with the original method in each examples and rectify the error that arises.
```p %w[ant bear cat].my_all? { |word| word.length >= 3 } #=> true```
```p %w[ant bear cat].my_all? { |word| word.length >= 4 } #=> false```
```p %w[ant bear cat].my_all?(/t/)                        #=> false```
```p [1, 2i, 3.14].my_all?(Numeric)                       #=> true```
```p [nil, true, 99].my_all?                              #=> false```
```p [].my_all?                                           #=> true```